In [31]:
def nameClass(val):
    if cls == 0:
                    print("Class Predicted: Normal")
    elif cls == 1:
                    print("Class Predicted: Unstable")

In [30]:
def nameClass(val):
    if cls == 0:
                    print("Class Predicted: Normal")
    elif cls == 1:
                    print("Class Predicted: Unstable")
import serial
import csv
from datetime import datetime
import keyboard
import pandas as pd
import pickle
from tensorflow.keras.models import load_model
# Load pre-trained model and scaler
model = load_model('my_model.h5')
clf = load_model('clf.h5')
with open('scaler.pkl', 'rb') as f:
    scaler = pickle.load(f)
serial_port_tr = 'COM7'
baud_rate = 9600
ser_tr = serial.Serial(serial_port_tr, baud_rate)
# Initialize DataFrame and counter
columns = ['Ax', 'Ay', 'Gx', 'Gy', 'Gz']
df = pd.DataFrame(columns=columns)
counter = 0
try:
    while True:
        line_tr = ser_tr.readline().decode('utf-8').strip()
        timestamp_trans = datetime.now().strftime("%H:%M:%S.%f")
        print(line_tr)
        values = line_tr[1:-1].split(',')
        if len(values) == 5:
            # Append values to DataFrame
            df.loc[counter] = values
            counter += 1
            # If length of DataFrame is 30, process and reset
            if counter == 30:
                print(df)
                # Transform data
                transformed_data = scaler.transform(df)
                # Ensure input shape matches model's input shape
                transformed_data = transformed_data.reshape((1, 30, 5))  # Adjust dimensions as needed
                # Predict using the model
                predictions = model.predict(transformed_data)
                cls = clf.predict(predictions)
                print("Predictions:", predictions)  
                print(f"class:" nameClass(cls))    
            if counter == 31:
                # Transform the last set of values (which didn't fill the DataFrame) and print
                transformed_values = scaler.transform(df.iloc[0].values.reshape(1, -1))
                print(f"Actual Values:{transformed_values}")

                # Reset counter and DataFrame
                counter = 0
                df = pd.DataFrame(columns=columns)
        if keyboard.is_pressed('a'):
            break
except KeyboardInterrupt:
    ser_tr.close()
    print("Serial ports closed.")


c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:329: UserWarning: Trying to unpickle estimator MinMaxScaler from version 1.4.1.post1 when using version 1.1.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(


1.04,1.26,-137.41,167.04,-222.01
0.14,0.34,295.06,229.24,449.05
0.88,0.93,-509.85,412.32,864.54
1.32,0.15,175.37,-533.84,972.20
1.11,-0.27,169.12,-61.77,286.25
1.75,-0.51,305.67,-382.50,-383.57
0.91,-0.00,-0.03,-0.03,-0.03
0.98,-0.37,-29.48,90.76,-121.37
0.98,-0.13,-147.23,33.69,-119.02
0.99,-0.11,-36.43,29.12,-179.30
1.01,-0.14,58.08,37.07,-287.10
1.54,-0.31,201.52,59.18,-572.90
0.87,0.66,26.19,45.55,-440.73
0.28,0.41,29.15,98.14,257.04
-0.06,0.83,-484.36,-95.00,859.39
1.04,0.37,-94.05,-347.77,884.51
1.24,-0.49,-95.21,-184.12,537.53
0.93,0.40,-201.46,264.45,-320.43
0.92,1.40,-275.27,395.76,-405.09
1.24,0.87,111.10,69.76,-175.73
0.94,0.24,298.17,34.12,-86.34
0.87,-0.13,87.26,80.15,-162.26
0.97,0.07,17.93,47.23,-227.84
1.07,0.14,20.37,10.58,-293.20
1.74,-0.42,120.49,-15.67,-607.62
1.37,1.17,-677.56,74.85,-415.21
0.16,0.17,-165.98,-175.73,362.32
0.69,0.78,-248.35,-123.90,966.71
1.33,0.06,-911.95,-734.70,790.67
1.11,-0.27,-31.07,118.32,-15.46
      Ax     Ay       Gx       Gy      Gz
0   

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


0.41,0.28,-597.10,-595.73,739.88
0.85,0.22,-679.70,-473.48,806.55
1.12,-0.59,-198.08,-197.38,662.16
1.02,0.14,-114.02,190.58,-44.27
2.00,0.33,210.73,-166.49,-472.62
1.05,-0.56,636.49,-239.05,-30.73
1.04,-0.69,269.27,128.29,-72.50
0.97,-0.14,3.78,83.78,-115.12
0.97,-0.04,22.13,61.43,-167.56
0.96,0.18,31.10,48.78,-189.12
1.02,-0.01,112.13,10.79,-230.98
1.63,-0.42,225.49,51.31,-511.01
1.45,1.04,-722.47,79.33,-346.92
0.20,0.18,27.32,-43.81,421.19
0.58,0.74,-448.41,-31.74,762.41
0.87,-0.05,-180.73,-528.02,971.52
1.14,-0.22,-75.95,16.98,387.20
1.97,-0.39,-396.10,-587.26,-407.90
1.08,-0.39,351.62,-118.51,-51.71
0.99,-0.36,51.59,135.40,-82.53
0.98,-0.19,-165.85,29.33,-97.84
     Ax     Ay       Gx       Gy      Gz
0   .73   0.27   186.80   347.23   -92.6
1   .97  -0.20    65.21    49.76   -71.8
2   .92   0.02    52.20    76.49  -112.6
3   .01  -0.01    -7.71    28.99  -158.0
4   .98   0.16    27.10     3.17  -187.2
5   .12   0.01    41.62    18.72  -299.6
6   .22  -0.37   102.16   -13.45  -530

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


0.64,-0.02,-412.41,-552.13,998.99
1.16,-0.08,-570.91,-442.99,845.15
0.98,-0.14,42.59,181.40,35.03
2.56,-0.28,-108.05,-824.91,-129.73
1.14,-0.42,513.51,-439.82,-101.74
1.05,-0.52,31.86,139.39,-87.59
0.97,-0.28,-150.21,15.21,-100.79
1.00,-0.26,-13.78,8.84,-210.64
1.06,-0.24,12.71,50.52,-304.42
1.49,-0.06,165.85,52.38,-508.78
0.77,0.61,-220.34,63.17,-375.46
0.27,0.27,59.45,-39.30,-137.84
0.41,0.75,188.08,-73.81,699.18
0.92,0.17,-229.15,-435.98,974.48
1.32,-0.73,-126.34,-231.68,701.43
0.93,0.35,-74.21,309.73,-372.84
0.05,1.49,-175.15,450.95,-503.02
1.09,0.43,-344.63,73.14,-207.77
0.96,0.28,-76.10,-4.15,-87.47
0.94,0.10,-100.91,40.49,-158.02
0.95,0.07,-26.71,17.13,-216.31
1.00,-0.07,145.52,-6.89,-338.87
1.61,-0.23,193.51,23.81,-589.85
1.12,1.06,-690.46,49.45,-237.62
0.38,0.17,155.73,-117.77,41.34
     Ax     Ay       Gx       Gy      Gz
0   .95   0.22    18.26    15.00  -224.6
1   .18  -0.09    78.41   -20.98  -394.7
2   .95  -0.30   300.76   -70.49  -610.2
3   .63   0.44  -319.54   179.76 

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


0.99,-0.25,19.15,28.05,-196.16
1.03,0.25,125.67,-19.33,-277.96
1.57,-0.29,-29.85,-56.01,-550.15
1.15,1.07,-862.62,9.33,-344.60
0.23,0.19,-281.22,-95.34,-19.21
0.49,0.48,-112.47,-426.19,556.19
0.60,0.65,-739.42,-673.02,845.70
1.47,-0.17,-736.65,70.21,698.26
0.92,0.40,-279.85,251.80,-329.66
0.85,1.42,-452.65,625.30,-324.21
1.01,0.79,-93.90,178.78,-164.45
0.93,0.24,207.56,-10.95,-58.69
0.92,-0.03,132.32,82.38,-115.21
1.02,0.01,23.96,21.13,-180.52
1.07,0.05,82.47,-44.24,-262.16
1.44,-0.04,-1.95,-24.18,-496.55
1.12,0.97,-386.49,-35.55,-336.10
0.30,0.14,-12.29,-135.21,-110.76
0.49,0.72,-267.77,-535.98,624.42
0.85,0.43,-457.56,-414.88,859.66
1.25,-0.22,-452.38,-147.74,647.44
0.95,0.30,-360.24,325.82,-347.59
0.38,1.02,-155.12,580.67,-523.20
1.18,0.45,-399.54,14.24,-184.30
     Ax     Ay       Gx       Gy      Gz
0   .41   0.35  -447.04  -938.51   998.9
1   .23  -0.79   -65.03   159.88   391.4
2   .78  -0.50  -473.05  -522.96   -33.6
3   .13   0.12   508.02  -200.85   -85.8
4   .92  -0.10   111

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


0.92,0.63,-364.09,109.73,-387.68
0.31,0.56,-338.05,-13.05,507.84
0.59,0.57,-525.37,-374.66,935.82
1.14,0.50,-598.93,-476.89,998.99
1.11,-0.26,-111.28,130.88,258.11
2.47,-0.89,-464.85,-299.97,-94.54
1.04,0.16,427.41,-148.11,-88.41
0.94,-0.24,146.62,129.48,-73.84
1.02,-0.19,-89.39,43.23,-76.28
0.96,-0.05,-57.38,41.52,-163.05
0.95,0.31,44.63,8.51,-192.41
1.12,-0.11,46.34,-19.30,-331.95
1.02,-0.24,467.56,-98.54,-557.77
0.73,0.31,-597.29,176.62,-440.98
0.47,0.69,48.93,-264.60,594.51
0.50,0.34,-580.24,-48.08,899.45
1.53,0.18,-667.84,-424.63,872.96
1.08,-0.41,-70.73,169.88,242.74
1.09,0.01,-106.31,315.67,-409.12
1.18,-1.24,-704.02,136.86,-114.09
0.90,-0.13,-320.49,102.29,-79.88
0.96,-0.11,-292.44,41.80,-126.55
0.94,-0.03,-177.47,22.16,-174.91
0.96,-0.01,-77.68,3.45,-214.57
1.04,0.07,-18.93,14.24,-337.93
1.21,-0.03,238.60,-3.11,-577.20
     Ax     Ay       Gx       Gy      Gz
0   .97  -0.08   -40.24    32.44  -154.6
1   .97   0.15    73.96    -4.39  -195.5
2   .07  -0.24    30.61    -1.59  -33

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


0.82,0.05,-301.01,281.83,-473.72
0.86,-0.94,-727.65,230.37,-232.50
0.92,-0.18,-332.71,22.04,-144.36
0.98,0.37,-171.68,45.91,-97.56
0.96,0.08,-110.03,31.37,-128.14
0.98,0.05,-26.13,9.66,-191.01
1.04,-0.24,99.12,10.98,-358.14
1.29,-0.25,91.37,-4.66,-592.99
1.18,0.88,-603.20,120.70,-235.37
0.33,0.45,-289.42,-133.23,399.70
0.46,0.33,-424.48,-354.79,773.35
0.98,0.27,-691.77,-263.29,898.29
1.26,0.03,-5.95,85.70,519.94
0.90,0.43,-417.01,292.07,-362.26
0.43,-0.05,-92.80,182.29,-477.74
0.99,0.12,-368.75,30.82,-130.55
0.94,0.25,-55.46,71.49,-89.12
0.93,0.16,-67.62,57.23,-148.08
0.95,0.21,-23.11,13.48,-186.25
0.99,-0.05,75.52,16.89,-275.70
1.51,-0.20,203.75,-3.51,-554.70
1.00,1.27,-779.97,-57.38,-227.65
0.16,0.16,-64.24,-192.50,-19.94
0.58,0.57,-474.39,-750.91,530.85
0.96,0.16,-354.39,-509.18,855.95
1.24,-0.50,-217.13,-237.23,734.94
     Ax     Ay       Gx       Gy      Gz
0   .35   0.21  -375.27  -114.36   381.1
1   .54   0.60  -282.53  -374.88   808.0
2   .93   0.23  -739.12  -530.98   986.2
3 

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


1.08,-0.20,-67.01,-16.74,-397.07
1.17,0.16,48.72,33.05,-617.99
0.53,0.33,7.26,-94.05,-268.20
0.55,0.64,-254.12,-69.21,494.39
1.24,0.67,-206.28,-300.37,735.67
0.98,-0.09,-751.65,-228.96,728.32
0.99,-0.04,-318.99,45.00,287.10
1.56,-0.62,150.49,98.14,-451.98
1.08,-1.06,-482.84,89.45,-92.10
1.00,0.08,-200.76,128.72,-62.77
0.93,0.06,-149.70,65.95,-96.89
0.94,0.08,-145.67,54.15,-138.90
0.93,0.18,-13.35,42.35,-176.49
1.02,-0.10,92.87,21.95,-317.41
1.50,-0.15,141.59,27.10,-610.49
1.30,1.22,-873.84,15.82,-283.93
0.23,0.09,-209.73,-154.12,305.88
0.41,0.61,-489.21,-186.86,777.80
1.06,0.40,-392.23,-429.09,983.75
1.22,-0.25,-124.73,39.60,485.06
1.41,-0.36,-313.23,246.86,-429.48
1.28,-0.73,-538.05,76.65,-143.75
0.93,-0.12,-276.19,117.44,-79.02
0.96,-0.02,-164.91,23.02,-94.12
0.96,-0.19,-36.04,-11.19,-203.38
     Ax     Ay       Gx       Gy      Gz
0   .71  -0.00  -999.02   394.51  -357.9
1   .81   1.10   530.46   291.98  -131.2
2   .93   0.24   371.55    34.63   -65.7
3   .96  -0.14   -64.48    57.3

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


0.62,0.11,-468.66,-239.88,687.53
1.16,0.02,-551.89,-335.82,778.66
1.11,0.34,-69.09,132.68,373.02
1.67,0.08,33.60,-176.68,-476.28
1.26,-0.54,216.68,38.72,44.18
1.05,-0.40,-117.32,118.45,-65.09
0.97,-0.20,-99.42,77.84,-82.50
0.97,-0.09,-88.14,24.66,-156.31
0.96,-0.17,14.66,13.87,-243.32
1.07,0.23,268.63,89.82,-419.97
0.90,-0.18,277.41,-247.74,-584.91
0.52,0.30,-331.65,129.57,-338.35
0.34,0.86,160.27,-205.09,656.77
0.74,0.15,-420.98,-574.27,864.33
1.19,-0.30,-741.80,-286.68,815.00
0.92,0.06,-246.07,209.70,51.68
2.10,0.22,-340.21,-462.90,-345.00
1.13,-0.25,639.33,-52.99,-93.93
0.88,-0.29,315.49,105.76,-103.90
0.98,-0.16,-90.00,3.99,-148.69
0.96,0.06,-54.02,-30.61,-205.03
1.09,0.12,19.79,-86.65,-260.55
1.30,-0.36,226.04,17.87,-480.95
1.19,0.89,-505.09,20.61,-275.37
0.51,0.24,183.29,-50.70,47.96
0.44,0.65,-243.72,-6.80,570.95
     Ax     Ay       Gx       Gy      Gz
0   .20  -0.27  -137.32    25.15  -404.9
1   .85   0.70  -229.15  -215.43  -487.0
2   .73   0.39  -244.15     7.93  -280.4
3   

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


1.10,0.36,-426.37,62.07,-110.67
0.98,0.46,117.29,67.99,-68.08
0.91,-0.03,63.63,52.35,-112.44
0.98,0.06,-29.57,16.52,-166.83
0.98,-0.04,56.49,-34.54,-237.59
1.27,-0.03,79.18,-21.98,-421.43
0.96,0.47,-494.12,-76.07,-348.23
0.54,0.33,22.93,-90.88,-135.09
0.64,0.81,-715.73,-221.59,410.91
0.80,0.29,-686.55,-433.48,605.06
1.11,-0.19,-312.93,-368.72,730.37
1.05,-0.26,-331.13,32.29,438.32
0.98,0.04,-483.75,338.75,-284.45
0.62,0.59,-536.86,625.95,-279.48
1.17,0.92,-360.58,93.75,-137.77
0.95,0.34,204.18,70.61,-105.64
0.91,0.13,119.39,92.68,-129.94
0.92,0.15,92.65,59.94,-186.89
1.04,0.09,52.38,5.85,-221.80
1.00,0.00,154.15,53.26,-384.15
0.95,-0.13,323.08,-86.40,-521.62
0.73,0.75,-376.89,46.68,-322.26
0.31,0.59,-132.74,-80.61,494.82
0.54,0.43,-501.10,-444.57,829.66
1.23,0.16,-624.48,-466.01,815.30
1.19,-0.08,-427.80,405.61,365.95
1.14,0.09,-260.85,155.61,-416.86
     Ax     Ay       Gx       Gy      Gz
0   .32  -0.07  -493.51  -285.55   593.5
1   .91   0.42  -158.38   274.51  -218.2
2   .43   0.73

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


1.53,-0.18,396.31,-36.04,-640.91
1.14,1.18,-999.02,49.48,-188.02
0.05,0.01,183.93,-387.04,13.41
0.50,0.57,-446.28,-243.72,636.40
0.92,0.35,-584.12,-344.36,694.15
1.16,-0.36,-74.36,-247.16,603.48
0.95,0.03,3.93,98.90,178.48
0.91,0.08,-100.82,301.74,-268.32
0.98,0.70,-641.07,464.66,-252.41
0.99,0.49,181.40,142.44,-121.37
0.99,-0.02,218.17,58.69,-110.27
0.96,0.04,61.71,92.96,-137.59
0.97,0.15,-24.24,14.15,-124.30
0.97,-0.00,29.91,-48.63,-142.65
1.16,-0.15,106.49,-64.48,-319.73
1.00,0.46,4.54,-57.01,-456.43
1.05,0.57,-472.35,19.88,-297.29
0.62,0.28,159.63,-46.52,163.14
0.70,0.51,-522.71,-140.73,566.83
0.98,0.31,-280.73,-435.73,917.71
1.29,-0.46,-95.70,-67.77,669.48
0.94,0.45,-501.31,328.05,-355.55
0.57,1.20,-709.51,220.03,-418.41
1.16,0.58,-218.87,28.69,-149.85
0.99,0.29,88.08,69.18,-48.29
     Ax     Ay       Gx       Gy      Gz
0   .94  -0.25  -308.75   161.19   -74.1
1   .94   0.17  -143.26    48.81   -86.4
2   .93   0.08   -83.29    22.59  -112.4
3   .96   0.10   -46.74    28.45  -175.

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


1.02,0.47,-378.90,-264.66,-472.87
0.34,0.34,-343.96,-152.59,-311.22
0.52,0.52,-415.03,-415.46,488.60
0.99,0.41,-333.75,-399.48,835.09
1.23,-0.48,-295.88,-339.82,727.80
1.01,0.08,-269.42,193.23,-6.40
2.16,-0.13,-68.14,-590.98,-125.06
1.15,-0.55,342.47,-92.62,-24.18
0.96,-0.62,-165.55,135.73,-98.57
0.95,0.01,-259.54,24.45,-69.88
0.94,-0.04,-145.85,0.49,-121.55
0.96,0.01,-4.30,-35.21,-176.37
1.01,-0.16,162.68,-67.71,-285.46
1.24,-0.17,325.79,-62.93,-489.33
1.14,0.89,-976.74,69.76,-238.60
0.35,0.19,-57.44,-222.10,-61.74
0.57,0.72,-508.29,-462.56,533.26
0.98,0.41,-381.01,-368.78,866.25
1.20,-0.33,-598.60,-182.53,587.93
0.96,0.25,-301.16,287.10,-236.19
0.71,0.63,-110.15,277.38,-428.57
0.99,0.21,-240.06,137.35,-67.13
0.99,0.27,29.66,33.57,-42.87
0.94,-0.08,0.79,48.08,-105.49
0.96,0.02,-17.13,36.55,-193.02
1.01,0.10,126.77,0.58,-258.14
1.51,-0.23,166.77,-42.99,-456.43
     Ax     Ay       Gx       Gy      Gz
0   .97   0.01  -108.02    18.41  -160.9
1   .94  -0.19    64.85   -45.58  -249.2
2   

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


1.31,-0.91,-263.23,-113.38,668.78
1.01,0.61,-115.12,353.17,-362.29
0.91,-1.18,-416.37,57.93,-221.19
0.88,-0.37,-436.10,96.74,-68.17
0.96,0.04,-261.74,64.18,-89.12
0.94,-0.00,-151.34,32.29,-143.90
0.95,0.19,-88.60,33.11,-161.25
0.97,-0.01,-0.79,-16.37,-225.58
1.42,0.04,91.77,3.81,-409.18
0.92,0.45,-14.51,-274.39,-474.79
0.35,0.24,-474.63,-39.02,-480.95
0.49,0.56,-538.29,-537.29,506.40
0.83,0.48,-616.95,-458.08,691.16
1.34,-0.38,-157.38,-317.59,798.63
1.00,0.08,-244.39,25.52,126.13
1.70,-0.45,247.68,-339.27,-288.51
1.04,-0.94,-32.90,-114.91,-93.14
0.96,-0.28,-245.55,111.52,-88.17
0.98,-0.19,-195.64,39.60,-101.62
0.96,-0.04,-153.26,49.82,-145.18
0.94,0.03,17.96,-20.00,-173.45
1.08,-0.08,207.01,-37.20,-367.84
1.21,-0.18,62.01,-41.13,-572.29
1.07,0.77,-529.05,123.90,-290.82
0.49,0.53,-431.71,-46.40,288.08
0.26,0.25,-397.35,-378.11,734.09
1.02,0.52,-708.81,-401.22,952.96
     Ax     Ay       Gx       Gy      Gz
0   .13  -0.02  -147.10  -223.54   -92.9
1   .34   0.64  -313.60  -318.23   569.4

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


0.99,0.07,101.34,-47.68,-234.91
1.59,-0.29,39.05,-55.46,-451.74
0.92,0.47,-22.90,-157.65,-518.35
0.61,0.32,-336.07,15.85,-231.86
0.34,0.66,-300.24,-229.36,589.51
0.90,0.74,-435.95,-515.91,998.99
1.38,-0.41,-821.13,-259.51,619.88
0.99,0.60,-104.51,316.43,-366.62
0.71,0.00,-184.85,-37.87,-222.87
1.05,0.08,-294.73,40.55,-31.43
0.89,0.09,-151.46,47.44,-27.04
1.00,-0.16,-100.21,45.37,-129.60
0.97,-0.01,-70.58,5.73,-204.36
0.91,-0.15,49.63,-23.23,-284.21
1.32,-0.33,177.41,-47.35,-533.02
0.98,0.97,-846.13,-80.24,-179.60
0.40,0.18,-291.68,-273.11,-89.30
0.56,0.31,-263.87,-413.78,536.04
0.95,0.49,-693.11,-277.84,772.50
1.30,-0.37,-184.82,-321.49,627.71
1.00,0.13,-169.45,191.40,-41.37
2.09,-0.29,-215.27,-501.71,32.65
1.15,-0.40,11.86,-121.40,-27.93
0.94,-0.28,-321.28,173.20,-120.70
0.99,0.14,-215.43,35.55,-106.71
     Ax     Ay       Gx       Gy      Gz
0   .56   0.11  -300.52  -141.52  -481.6
1   .20  -0.61    15.64   125.98   -52.9
2   .00  -0.28   -78.02   116.65  -125.0
3   .97  -0.07  -163.

c:\Users\Shubham Kumar Jha\AppData\Local\Programs\Python\Python310\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but MinMaxScaler was fitted with feature names
  warnings.warn(


0.34,0.34,-110.82,-237.50,-84.30
0.32,0.03,-245.88,-321.59,558.26
0.90,0.57,-558.63,-339.79,801.68
1.12,-0.19,-441.62,-206.59,585.09
0.95,-0.09,-26.40,129.30,36.59
1.13,-0.33,-136.46,125.27,-368.66
1.08,-0.51,-812.93,83.29,-99.91
0.96,0.20,-267.99,206.04,-69.94
0.91,0.13,-46.16,64.02,-140.64
0.95,0.22,-103.93,53.11,-149.66
0.96,0.05,-16.80,-25.58,-177.38
1.04,-0.21,213.02,-58.23,-347.50
1.21,-0.37,229.88,-22.87,-572.84
Serial ports closed.


0.64,-0.02,-412.41,-552.13,998.99
1.16,-0.08,-570.91,-442.99,845.15
0.98,-0.14,42.59,181.40,35.03
2.56,-0.28,-108.05,-824.91,-129.73
1.14,-0.42,513.51,-439.82,-101.74
1.05,-0.52,31.86,139.39,-87.59
0.97,-0.28,-150.21,15.21,-100.79
1.00,-0.26,-13.78,8.84,-210.64
1.06,-0.24,12.71,50.52,-304.42
1.49,-0.06,165.85,52.38,-508.78
0.77,0.61,-220.34,63.17,-375.46
0.27,0.27,59.45,-39.30,-137.84
0.41,0.75,188.08,-73.81,699.18
0.92,0.17,-229.15,-435.98,974.48
1.32,-0.73,-126.34,-231.68,701.43
0.93,0.35,-74.21,309.73,-372.84
0.05,1.49,-175.15,450.95,-503.02
1.09,0.43,-344.63,73.14,-207.77
0.96,0.28,-76.10,-4.15,-87.47
0.94,0.10,-100.91,40.49,-158.02
0.95,0.07,-26.71,17.13,-216.31
1.00,-0.07,145.52,-6.89,-338.87
1.61,-0.23,193.51,23.81,-589.85
1.12,1.06,-690.46,49.45,-237.62
0.38,0.17,155.73,-117.77,41.34
     Ax     Ay       Gx       Gy      Gz
0   .95   0.22    18.26    15.00  -224.6
1   .18  -0.09    78.41   -20.98  -394.7
2   .95  -0.30   300.76   -70.49  -610.2
3   .63   0.44  -319.54   179.76  -314.1
4   .18   0.84   -22.56  -223.60   684.4
5   .64  -0.02  -412.41  -552.13   998.9
6   .16  -0.08  -570.91  -442.99   845.1
7   .98  -0.14    42.59   181.40    35.0
8   .56  -0.28  -108.05  -824.91  -129.7
9   .14  -0.42   513.51  -439.82  -101.7
10  .05  -0.52    31.86   139.39   -87.5
11  .97  -0.28  -150.21    15.21  -100.7
12  .00  -0.26   -13.78     8.84  -210.6
13  .06  -0.24    12.71    50.52  -304.4
14  .49  -0.06   165.85    52.38  -508.7
15  .77   0.61  -220.34    63.17  -375.4
16  .27   0.27    59.45   -39.30  -137.8
17  .41   0.75   188.08   -73.81   699.1
18  .92   0.17  -229.15  -435.98   974.4
19  .32  -0.73  -126.34  -231.68   701.4
20  .93   0.35   -74.21   309.73  -372.8
21  .05   1.49  -175.15   450.95  -503.0
22  .09   0.43  -344.63    73.14  -207.7
23  .96   0.28   -76.10    -4.15   -87.4
24  .94   0.10  -100.91    40.49  -158.0
25  .95   0.07   -26.71    17.13  -216.3
26  .00  -0.07   145.52    -6.89  -338.8
27  .61  -0.23   193.51    23.81  -589.8
28  .12   1.06  -690.46    49.45  -237.6
29  .38   0.17   155.73  -117.77    41.3
1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 90ms/step
1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 81ms/step
Predictions: [[0.782868   0.3395939  0.5409777  1.112203   0.36276412]]
Class: Stable
1.16,0.56,22.62,-45.15,676.86
Actual Values:[[ 0.86013986 -0.74        0.61080733  0.37996095  0.28862345]